lib/autoproj/daemon/package_repository.rb

```diff
@@ -46,7 +47,13 @@ def repo_url
 
             # @return [String]
             def branch
-                vcs[:remote_branch] || vcs[:branch] || "master"
+                current_importer = autobuild.importer
+
+                vcs[:remote_branch] ||
+                    vcs[:branch] ||
+                    current_importer.try_resolve_remote_head_from_server(
```

Comment:
1. Use `Autobuild::Git#resolve_remote_head` instead.
2. Don't create the importer unless you have to (if both `vcs[:remote_branch]` and `vcs[:branch]` are unset)